apps/src/accounts/AccountInformation/index.tsx

```diff
@@ -0,0 +1,393 @@
+          },
+        };
+      } else {
+        error = new Error('Unexpected failure: ' + response.status);
+      }
+      // error is handled in ChangeEmailModal
+      throw error;
+    }
+
+    // update succeeded, hide modal and update email state with the new email
+    setShowChangeEmailModal(false);
+    setEmail(newEmail);
+  };
+
+  return (
+    <>
+      <hr />
+      <Heading2 visualAppearance="heading-sm" className={styles.sectionHeader}>
+        Account information
+      </Heading2>
+      <form>
+        <div className={styles.inputContainer}>
+          {/* verified teacher account */}
+          {authorizedTeacher && (
+            <div className="field">
+              <label className={styles.authorizedTeacher}>
+                ✔ {i18n.accountInformation_verified_teacher()}
+              </label>
+            </div>
+          )}
+
+          {/* display name */}
+          <TextField
+            className={styles.input}
+            label={i18n.accountInformation_display_name()}
+            onChange={e => setName(e.target.value)}
+            value={name}
+            name="user[name]"
+            readOnly={secretPictureAccountOnly}
+            maxLength={255}
+            helperMessage={displayNameHelperMessage}
+            errorMessage={errors.name?.[0]}
+          />
+
+          {/* username */}
+          {userUsername && (
+            <TextField
+              className={styles.input}
+              label={i18n.accountInformation_username()}
+              onChange={e => setUsername(e.target.value)}
+              value={username}
+              name="user[username]"
+              maxLength={20}
+              minLength={5}
+              errorMessage={errors.username?.[0]}
+            />
+          )}
+
+          {/* email */}
+          {!teacherManagedAccount && !parentManagedAccount && (
+            <>
+              <TextField
+                className={classNames(styles.input, styles.emailInput)}
+                label={i18n.accountInformation_email()}
+                helperMessage={
+                  migrated && !isStudent
+                    ? i18n.accountInformation_email_hint()
+                    : undefined
+                }
+                readOnly={true}
+                onChange={() => {}}
+                value={emailValue}
+                name="user[email]"
+                errorMessage={errors.email?.[0]}
+              />
+              {/* TODO: replace Link with design system semantic button styled as an inline link when it exists */}
+              {shouldSeeEditEmailLink && (
+                <Link
+                  role="button"
+                  href="#"
+                  text={i18n.updateEmail()}
+                  onClick={e => {
+                    e.preventDefault();
+                    setShowChangeEmailModal(true);
+                  }}
+                  className={styles.link}
+                  size="s"
+                />
```

Comment:
I think we can use our `Button` design component which takes a `styleAsText` parameter. Or are we waiting on a newer one?

Reply:
yeah... I had a convo with moshe and denys about this. It looks like the Button with styleAsText is a legacy component and there's not an equivalent in the design system yet 😞 

I'm not stoked to use an `a` tag with a button role though, so I added that TODO comment. what do you think, should I swap out the semantically wrong DSCO component for a deprecated legacy component until the new one is ready? I'm on the fence tbh 🤷

Reply:
Ahh gotcha gotcha, works for me!